File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int count = nums.size()/2;
    sort(nums.begin(), nums.end());
    //중복제거
    nums.erase(unique(nums.begin(),nums.end()),nums.end());
    
    if(count<nums.size()) answer=count;
    else answer=nums.size();
    
    return answer;
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashMap<Integer, Integer> hash = new HashMap<>();
        
        for(int num: nums){
            if(!hash.containsKey(num)){
                hash.put(num, 1);
            }
        }
        
            
        
        return Math.min(nums.length/2, hash.size());
    }
}